staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -480,12 +485,7 @@ func (g *Cloud) getInstancesByNames(names []string) ([]*gceInstance, error) {
 			}
 		}
 		klog.Errorf("Failed to retrieve instances: %v", failed)
-		return nil, cloudprovider.InstanceNotFound
-	}
-
-	var ret []*gceInstance
-	for _, instance := range found {
-		ret = append(ret, instance)
+		return ret, cloudprovider.InstanceNotFound
```

Comment:
I'm wondering if it won't break some assuming somewhere... but hopefully not.

Reply:
oh I see now that this was modified. It feels weird to me to both return an error and a partial result back, then do some intense error checking to surface messages on the caller side.

What are the cases where an instance may not exist and why would we want to not just throw and error and let the caller of the `BulkVerify` function retry with the correct instances/nodes next time?

Reply:
I think the intention here was to avoid situation when we list say 5k instances, 1 is not found and we drop everything.

But after thinking a bit more about it, I would much rather prefer refactoring this function (in the worst case introducing a new one) that would be returning some structured thing (rather than partial result and error).

So something like map[string]*gce.Instance (and treat nil as notFound) or map[string]struct{gce.instance, error}

Would be much more intuitive.

Reply:
I didn't want to make big changes to what this function returns because it is staged to a separate repo that can be used by many other people. What you suggest makes more sense, but I fear is not backwards compatible enough.

Reply:
Would it be strange to create a new function?

Reply:
This new function would be very similar to the old one. I can see the new one being a wrapper on the old function, but I don't know if it makes sense to have one just to change the return value. With the current return, I do infer which instances the InstanceNotFound error refers to by cross listing it with the instances we expect.

Reply:
I would prefer there be a new function that's very similar. The old function could be a wrapper on the new function or both could be wrapper on a third function. Implementation is up to you but I don't think that we should be changing the guarantees of the original function (especially to something that is non-standard like partial results while throwing an error).

Reply:
I don't believe this function is exported so it can't be consumed elsewhere. So you could potentially just change the signature of this function and update all the callers

Reply:
This is a private method and we are free to do whatever we want with it.
Please create a separate PR to change it to what we need (what I suggested above).